protokt-codegen/src/main/resources/services.stg

```diff
@@ -13,30 +13,60 @@
  * limitations under the License.
  */
 
-service(name, type, in, out, deprecated, ops, methods, descriptors) ::= <<
+import "renderers.stg"
 
+service(name, qualifiedName, descriptor, methods) ::= <<
 object <name>Grpc {
-    <name>
-    <type>
-    <in>
-    <out>
-    <deprecated>
-    <ops>
-    <methods>
-    <descriptors>
-}
->>
+    const val SERVICE_NAME = "<qualifiedName>"
 
-methodDesc(name, type, is, os, in, out) ::= <<
-    val METHOD_<name> =
-            io.grpc.MethodDescriptor.newBuilder\<<in>, <out>\>()
+    val serviceDescriptor: ServiceDescriptor by lazy {
+        <descriptor>
+    }
 
->>
+    <methods : { m | val <m.lowerName>Method: MethodDescriptor\<<m.in>, <m.out>\> by lazy {
+    <m.body><\n><\\>
+    \}}; separator="\n\n">
 
-descriptorSupplier(fqcn) ::= <<
+    @JvmName("descriptor")
+    @JvmSynthetic
+    fun getServiceDescriptor() =
```

Comment:
would `@get:JvmName` work?

Reply:
`This annotation is not applicable to target 'member function' and use site target '@get'`

Reply:
Strictly speaking the `@JvmSynthetic` annotation isn't need for this to compile, it's just to hide it so the getter is the only visible item from Java.

Reply:
I meant `@get:JvmName`  on the `val`

Reply:
Ah - the catch is grpc-kotlin calls the `get()` functions _from Kotlin_, so it has to be present as a function in Kotlin.

Reply:
https://github.com/grpc/grpc-kotlin/blob/master/compiler/src/test/java/io/grpc/kotlin/generator/HelloWorldProtoGrpcKt.expected#L57

Reply:
So yes - `@get:JvmName` on the `val`s works, but then you have an extra visible member from Java. I don't know whether we care.